spacy/pipeline/pipes.pyx

```diff
@@ -1228,13 +1234,30 @@ class EntityLinker(Pipe):
                 # the gold annotations should link to proper entities - if this fails, the dataset is likely corrupt
                 if not (start, end) in ents_by_offset:
                     raise RuntimeError(Errors.E188)
+
                 ent = ents_by_offset[(start, end)]
 
                 for kb_id, value in kb_dict.items():
                     # Currently only training on the positive instances
                     if value:
                         try:
-                            sentence_docs.append(ent.sent.as_doc())
+                            # find the sentence in the list of sentences.
+                            sent_index = sentences.index(ent.sent)
+                            
+                            # get n previous sentences, if there are any
+                            start_sentence = max(0, sent_index - self.n_sents)
+
+                            # get n posterior sentences, or as many < n as there are
+                            end_sentence = min(len(sentences) -1, sent_index + self.n_sents)
+
+                            # get token positions
+                            start_token = sentences[start_sentence].start
+                            end_token = sentences[end_sentence].end
+
+                            # append that span as a doc to training
+                            sent_doc = doc[start_token:end_token].as_doc()
+                            sentence_docs.append(sent_doc)
+
                         except AttributeError:
```

Comment:
Can this `except` block be moved above? Ideally I would want:

```
try: 
    sent = ent.sent
except AttributeError:
    ...
```

try/except blocks should be scoped as narrowly as possible, ideally over a single statement so that they don't catch accidentally.